docs/03-declarative-layout.md

```diff
@@ -29,7 +28,116 @@ configurable rules.
 
 @DOC_layout_WIDGET_LIST@
 
-## Placing widgets
```

Comment:
If you want, I can produce a fixup commit for the following comments, although some of them are questions instead of just plain suggestions.